Unprefix -webkit-ruby-position

anttijk opened the pull request:
#### ba5aedf5f753d40e7f3d45e56105762b7408d9ea
<pre>
Unprefix -webkit-ruby-position
<a href="https://bugs.webkit.org/show_bug.cgi?id=151306">https://bugs.webkit.org/show_bug.cgi?id=151306</a>
<a href="https://rdar.apple.com/86128259">rdar://86128259</a>

Reviewed by Alan Baradlay.

Support `ruby-position` with the standard value names (over/under/inter-character) excluding &apos;alternate&apos;.

* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-ruby/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-ruby/line-spacing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-ruby/parsing/ruby-position-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002-expected.txt:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::toCSSValueID):
(WebCore::fromCSSValueID):
* Source/WebCore/css/CSSProperties.json:

Add `ruby-position` and mark `-webkit-ruby-position` as its cascade alias.

Use the new cascade-alias mechanism as a simple alias does not work because the syntax differs
and the related-property mechanism can&apos;t be used as the property is marked top-priority.

The new mechanism replaces the existing &apos;synonym&apos; mechanism that does not have correct behavior
when both synonyms are specified (the latter should override the earlier one). It was only used
by one property.

Also use the mechanism for -webkit-mask-source-type.

* Source/WebCore/css/CSSToStyleMap.cpp:
(WebCore::CSSToStyleMap::mapFillMaskMode):
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle const):
* Source/WebCore/css/process-css-properties.py:
(StylePropertyCodeGenProperties):
(StylePropertyCodeGenProperties.from_json):
(StyleProperty.perform_fixups_for_cascade_alias_properties):
(StyleProperty.perform_fixups):
(GenerateCSSPropertyNames):

Add support for cascade-alias mechanism.

* Source/WebCore/layout/formattingContexts/inline/InlineFormattingUtils.cpp:
(WebCore::Layout::InlineFormattingUtils::textEmphasisForInlineBox):
* Source/WebCore/layout/formattingContexts/inline/ruby/RubyFormattingContext.cpp:
(WebCore::Layout::rubyPosition):
(WebCore::Layout::RubyFormattingContext::placeAnnotationBox):
(WebCore::Layout::RubyFormattingContext::adjustLayoutBoundsAndStretchAncestorRubyBase):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::conservativelyCollectChangedAnimatableProperties const):
* Source/WebCore/rendering/style/RenderStyleConstants.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/rendering/style/RenderStyleConstants.h:

Rename to match the standard naming.

* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::initialRubyPosition):
* Source/WebCore/style/PropertyCascade.cpp:
(WebCore::Style::PropertyCascade::addMatch):
* Tools/Scripts/webkitpy/style/checkers/jsonchecker.py:
(JSONCSSPropertiesChecker.check_codegen_properties):

Canonical link: <a href="https://commits.webkit.org/281804@main">https://commits.webkit.org/281804@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eb1f51d96a5156610d277736b3ebf876945df56c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/61055 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/40416 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13636 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64992 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11604 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/63185 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/48093 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11879 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/49352 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/60/builds/8060 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/63089 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37611 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/52895 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30182 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/60580 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/34290 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/10119 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/10516 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/56107 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/10415 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66722 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/5002 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/10230 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/56721 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/5024 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52858 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56914 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13617 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/4143 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/36220 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/37303 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/38397 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/37047 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/process-css-properties.py:
```diff
@@ -716,6 +721,16 @@ def perform_fixups_for_related_properties(self, all_properties):
+                raise Exception(f"Property {self.name} is a parser alias for an unknown property: {self.codegen_properties.parser_alias}.")
+
+            if not self.codegen_properties.skip_builder:
+                raise Exception(f"Property {self.name} is a parser alias and shoud also set skip-builder.")
```
```suggestion
                raise Exception(f"Property {self.name} is a parser alias and should also set skip-builder.")
```

nt1m commented on Source/WebCore/css/process-css-properties.py:
```diff
@@ -743,6 +758,7 @@ def perform_fixups(self, all_properties):
         self.perform_fixups_for_synonyms(all_properties)
```
Shouldn't we remove the code for synonyms?

anttijk replied:
yeah, later

nt1m reviewed (commented):
There's a couple more properties that could be aliases: https://searchfox.org/wubkat/search?q=%22related-property%22%3A&path=&case=false&regexp=false

(though I guess it could be a follow up)

anttijk committed:
Unprefix -webkit-ruby-position
https://bugs.webkit.org/show_bug.cgi?id=151306
rdar://86128259

Reviewed by Alan Baradlay.

Support `ruby-position` with the standard value names (over/under/inter-character) excluding 'alternate'.

* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-ruby/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-ruby/line-spacing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-ruby/parsing/ruby-position-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002-expected.txt:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::toCSSValueID):
(WebCore::fromCSSValueID):
* Source/WebCore/css/CSSProperties.json:

Add `ruby-position` and mark `-webkit-ruby-position` as its cascade alias.

Use the new cascade-alias mechanism as a simple alias does not work because the syntax differs
and the related-property mechanism can't be used as the property is marked top-priority.

The new mechanism replaces the existing 'synonym' mechanism that does not have correct behavior
when both synonyms are specified (the latter should override the earlier one). It was only used
by one property.

Also use the mechanism for -webkit-mask-source-type.

* Source/WebCore/css/CSSToStyleMap.cpp:
(WebCore::CSSToStyleMap::mapFillMaskMode):
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle const):
* Source/WebCore/css/process-css-properties.py:
(StylePropertyCodeGenProperties):
(StylePropertyCodeGenProperties.from_json):
(StyleProperty.perform_fixups_for_cascade_alias_properties):
(StyleProperty.perform_fixups):
(GenerateCSSPropertyNames):

Add support for cascade-alias mechanism.

* Source/WebCore/layout/formattingContexts/inline/InlineFormattingUtils.cpp:
(WebCore::Layout::InlineFormattingUtils::textEmphasisForInlineBox):
* Source/WebCore/layout/formattingContexts/inline/ruby/RubyFormattingContext.cpp:
(WebCore::Layout::rubyPosition):
(WebCore::Layout::RubyFormattingContext::placeAnnotationBox):
(WebCore::Layout::RubyFormattingContext::adjustLayoutBoundsAndStretchAncestorRubyBase):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::conservativelyCollectChangedAnimatableProperties const):
* Source/WebCore/rendering/style/RenderStyleConstants.cpp:
(WebCore::operator<<):
* Source/WebCore/rendering/style/RenderStyleConstants.h:

Rename to match the standard naming.

* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::initialRubyPosition):
* Source/WebCore/style/PropertyCascade.cpp:
(WebCore::Style::PropertyCascade::addMatch):
* Tools/Scripts/webkitpy/style/checkers/jsonchecker.py:
(JSONCSSPropertiesChecker.check_codegen_properties):

Canonical link: https://commits.webkit.org/281804@main
